### PR TITLE
Fixes #1757

### LINKED ISSUE #1757: bigip_device_info:  gather_subset:  - client-ssl-profiles does not return private key from cert-key-chain as per CLI
##### ISSUE TYPE

 - Bug Report
##### ANSIBLE VERSION

```
ansible 2.9.7
  config file = /app/ansible.cfg
  configured module search path = ['/app/module']
  ansible python module location = /usr/local/lib/python3.6/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.6.8 (default, Apr  2 2020, 13:34:55) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```

##### PYTHON VERSION

```
Python 3.6.8
```

##### BIGIP VERSION

```
Sys::Version
Main Package
  Product     BIG-IP
  Version     13.1.3
  Build       0.0.6
  Edition     Final
  Date        Fri Aug  2 20:39:15 PDT 2019
```

##### CONFIGURATION

n/a

##### OS / ENVIRONMENT

CentOS Linux release 7.8.2003 (Core)

##### SUMMARY

The returned client_ssl_profiles subset does **not** contain the key file name for a client ssl profile.

##### STEPS TO REPRODUCE

To reproduce the problem, create a task in a playbook to return the client ssl profiles subset, and parse/review the output.  The cert-key-chain info does not display the key, as expected

Key is not listed in the [ClientSslProfilesParameters(BaseParameters)](https://github.com/F5Networks/f5-ansible/blob/devel/library/modules/bigip_device_info.py#L8054) of bigip_device_info.py.

```
    - name: "Gather BigIP F5 Client SSL Profiles"
      bigip_device_info:
        provider: "{{ provider }}"
        gather_subset:
          - client-ssl-profiles
      register: bigip_facts

    - name: "Display BigIP F5 Client SSL Profiles"
       debug:
         msg: "{{ bigip_facts }}"

```

##### EXPECTED RESULTS

From the command line
```
ltm profile client-ssl my-test-client-cssl {
< snipped>
    cert-key-chain {
        default {
            cert my-test-client.crt
            chain my-test-client-chain.crt
            key my-test-client.key
        }
    }
< snipped >
}
```
From the Ansible module

```
  {
    "modssl_methods": "no",
    "renegotiation_maximum_record_delay": 0,
    "forward_proxy_lookup_by_ipaddr_port": "no",
    "cache_timeout": 3600,
    "strict_resume": "no",
    "description": "none",
    "chain_file": "/Common/my-test-client-chain.crt",
    "handshake_timeout": 10,
    "renegotiation": "yes",
    "certificate_file": "/Common/my-test-client.crt",
    "key_file": "/Common/my-test-client.key"       ###    <======= expected something like this
    "sni_default": "no",
    "forward_proxy_enabled": "no",
    "forward_proxy_certificate_extension_include": [
      "basic-constraints",
      "subject-alternative-name"
    ],
    "sni_require": "no",
    "allow_non_ssl": "no",
    "parent": "/Common/test.app/default",
    "ciphers": ["DEFAULT:!3DES:!DHE"],
    "secure_renegotiation_mode": "require",
    "retain_certificate": "yes",
    "renegotiation_period": 0,
    "name": "my_test_cssl",
    "alert_timeout": 0,
    "profile_mode_enabled": "yes",
    "authenticate_frequency": "once",
    "session_ticket": "no",
    "cache_size": 262144,
    "unclean_shutdown": "yes",
    "authenticate_depth": 9,
    "peer_certification_mode": "ignore",
    "forward_proxy_certificate_lifespan": 30,
    "full_path": "/Common/my_test_cssl"
  },
```
##### ACTUAL RESULTS

Key file name not mentioned
```
{
    "modssl_methods": "no",
    "renegotiation_maximum_record_delay": 0,
    "forward_proxy_lookup_by_ipaddr_port": "no",
    "cache_timeout": 3600,
    "strict_resume": "no",
    "description": "none",
    "chain_file": "/Common/my-test-client-chain.crt",
    "handshake_timeout": 10,
    "renegotiation": "yes",
    "certificate_file": "/Common/my-test-client.crt",
    "sni_default": "no",
    "forward_proxy_enabled": "no",
    "forward_proxy_certificate_extension_include": [
      "basic-constraints",
      "subject-alternative-name"
    ],
    "sni_require": "no",
    "allow_non_ssl": "no",
    "parent": "/Common/test.app/default",
    "ciphers": ["DEFAULT:!3DES:!DHE"],
    "secure_renegotiation_mode": "require",
    "retain_certificate": "yes",
    "renegotiation_period": 0,
    "name": "my_test_cssl",
    "alert_timeout": 0,
    "profile_mode_enabled": "yes",
    "authenticate_frequency": "once",
    "session_ticket": "no",
    "cache_size": 262144,
    "unclean_shutdown": "yes",
    "authenticate_depth": 9,
    "peer_certification_mode": "ignore",
    "forward_proxy_certificate_lifespan": 30,
    "full_path": "/Common/my_test_cssl"
  },
```

### PR DESCRIPTION
**Pull request by nitinthewiz**

### CODE REVIEW

File: test/integration/targets/bigip_device_info/tasks/issue-01757.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+
+- name: Issue 01757 - Select all facts, single keyword
+  f5networks.f5_modules.bigip_device_info:
```
Comment (wojtek0806) [inline test/integration/targets/bigip_device_info/tasks/issue-01757.yaml:4]: why is this here? this should work without the FQCN to the collection

File: test/integration/targets/bigip_device_info/tasks/issue-01757.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    gather_subset:
+      - client-ssl-profiles
+  register: bigip_facts
+
+- name: "Display BigIP F5 Client SSL Profiles"
+  debug:
+    msg: "{{ bigip_facts }}"
```
Comment (wojtek0806) [inline test/integration/targets/bigip_device_info/tasks/issue-01757.yaml:11]: this is not how you should check if the right data was grabbed, use asserts and jmespath filters to find out if the data was correct, for consistency you can create resources to check for.

File: test/integration/targets/bigip_device_info/tasks/issue-01757.yaml (hunk 1 above)
Comment (nitinthewiz) [reply test/integration/targets/bigip_device_info/tasks/issue-01757.yaml:4]: For some reason it wasn't... I'll remove it, of course. Need to investigate my env and see why it's working...

File: test/integration/targets/bigip_device_info/tasks/issue-01757.yaml (hunk 2 above)
Comment (nitinthewiz) [reply test/integration/targets/bigip_device_info/tasks/issue-01757.yaml]: the first element will always be the clientssl default profile, which can't be removed from the BIGIP, so I'll just check against that.

File: test/integration/targets/bigip_device_info/tasks/issue-01757.yaml (hunk 1 above)
Comment (wojtek0806) [reply test/integration/targets/bigip_device_info/tasks/issue-01757.yaml]: check the email
i sent on the dev env few days ago if this is all good and still broken ill test locally when i am back

File: test/integration/targets/bigip_device_info/tasks/issue-01757.yaml (hunk 2 above)
Comment (wojtek0806) [reply test/integration/targets/bigip_device_info/tasks/issue-01757.yaml]: if the defaul data changes between versions it might fail its better to have something uniform across versions

**Comment by nitinthewiz** [comment]
> if key_file is not in returnables, put it there with proper description

Added to the docs

File: test/integration/targets/bigip_device_info/tasks/issue-01757.yaml (hunk 1 above)
Comment (nitinthewiz) [reply test/integration/targets/bigip_device_info/tasks/issue-01757.yaml]: No no, it's all good... Somehow started working... I think it's because I logged into bash again...

File: test/integration/targets/bigip_device_info/tasks/issue-01757.yaml (hunk 2 above)
Comment (nitinthewiz) [reply test/integration/targets/bigip_device_info/tasks/issue-01757.yaml]: Hasn't changed in 15 years.

Comment (nitinthewiz) [reply test/integration/targets/bigip_device_info/tasks/issue-01757.yaml]: Fair enough, I'll add a create a resource and check against it.

Comment (nitinthewiz) [reply test/integration/targets/bigip_device_info/tasks/issue-01757.yaml]: Added the clientssl profile creation and verification.
